### DISCUSSION TITLE
How SHOULD we delete PackageQuery and Install links with contains?

### DISCUSSION THREAD

**Discussion by FreePhoenix888** [deep-foundation/Discussions#14] [Q&A] [answered]
@Konstantin Dyachenko#8793 , if we delete PackageQuery and Install this way:
```ts
await deep.delete({
   up: {
      tree_id: {
         _id: ['@deep-foundation/core', 'containTree']
      },
      parent: {
         type_id: {
            _id: ['@deep-foundation/core', 'Contain']
         },
         to: {
            _or: [
               {
                  type_id: {
                     _id: ['@deep-foundation/npm-packager', 'Install']
                  }
               },
               {
                  type_id: {
                     _id: ['@deep-foundation/core', 'PackageQuery']
                  }
               }
            ]
         }
      }
   }
})
```
Then the packages that were installed by using Install link will be deleted too. How can we avoid this? How SHOULD we delete only PacakgeQuery and Install links and contains to then?
Note that my question is about how we SHOULD do it, not ways to do it

**Comment by konard** [comment, marked as answer]
No, delete down by contain tree does not delete `Package` links, at least when these are created by Packager UI. I have no problem deleting `PackageQuery` and Install links by using delete down on two contains - to Install and to `PackageQuery` links.

I suggest creating a query, that shows all links after its creation. Then you should install the smallest possible package and carefully watch what links are created. Contain tree may delete only links in that tree, but there are more links between Install and Package which will be kept as invisible garbage.

**Reply by FreePhoenix888** [reply]
It is true. I have tried to reproduce but the code above does not delete packages. Nice!
